src/__tests__/core/web_api/p2_api.test.js

```diff
@@ -37,7 +38,22 @@ describe('Auth0APIClient', () => {
   });
   describe('init', () => {
     describe('with overrides', () => {
+      it('uses telemetry set in the `auth0Client` query param if available', () => {
+        const telemetryIn = { fromTest: true, env: { envOverride: true } };
```

Comment:
@luisrudge I think we should whitelist the params we allow in the telemetry. e.g. only allow to change the ones we will need on the backend. @joshcanhelp Any strong feelings on this or just leave it as it is?

Reply:
this wasn't in the RFC, so it's not here 😬 ideally, people should be able to use whatever telemetry they want, right?

Reply:
> ideally, people should be able to use whatever telemetry they want, right?

Ideally, no, but I'm not sure how far we need to go towards ensuring what is sent. I, personally, don't like the idea of opening up the telemetry to whatever they want to send, would rather it restrict to what's parsed (`name`, `version`, `env`) but processing on the data side will only look at those properties.

Technically, anyone **CAN** send anything they want as a header/query param outside of our SDKs. To be honest, I don't know how that will appear in the logs. In the actual reports, though, only specific properties will appear.

All of that said ... I would not use untracked properties in the tests, no reason to do that. Just set a `name` of `test-sdk` or similar and a version. If for some reason we whitelist properties, this would fail unnecessarily. Also, bad example if someone is grepping for examples.

Reply:
@joshcanhelp I updated the test props.